src/rosdep2/rospkg_loader.py

```diff
@@ -117,6 +117,33 @@ def get_loadable_resources(self):
             self._loadable_resource_cache = list(self._rospack.list())
         return self._loadable_resource_cache
 
+    def dep_version_checking(self, deps):
+        names_with_version = []
+        for i in (0, len(deps) - 1):
+            version_values = []
+            try:
+                version_values.append(getattr(deps[i], "version_eq"))
+                version_values.append(getattr(deps[i], "version_gt"))
+                version_values.append(getattr(deps[i], "version_gte"))
+                version_values.append(getattr(deps[i], "version_lt"))
+                version_values.append(getattr(deps[i], "version_lte"))
+            except:
```

Comment:
You should catch the specific error that is expected instead of catching all errors.